src/DGtal/topology/NeighborhoodConfigurations.h

```diff
@@ -0,0 +1,118 @@
+/**
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ **/
+
+/**
+ * @file NeighborhoodConfigurations.h
+ *
+ * @author Pablo Hernandez-Cerdan. Institute of Fundamental Sciences.
+ * Massey University. Palmerston North, New Zealand
+ *
+ * @date 2016/03/25
+ *
+ * Generation of Look Up Tables for predicates function in a voxel
+ * neighborhood.
+ *
+ * This file is part of the DGtal library.
+ */
+
+#if defined(NeighborhoodConfigurations_RECURSES)
+#error Recursive header files inclusion detected in NeighborhoodConfigurations.h
+#else // defined(NeighborhoodConfigurations_RECURSES)
+/** Prevents recursive inclusion of headers. */
+#define NeighborhoodConfigurations_RECURSES
+
+#if !defined NeighborhoodConfigurations_h
+/** Prevents repeated inclusion of headers. */
+#define NeighborhoodConfigurations_h
+//////////////////////////////////////////////////////////////////////////////
+// Inclusions
+#include <iostream>
+#include <bitset>
+#include <unordered_map>
+#include "boost/dynamic_bitset.hpp"
+#include <DGtal/base/CountedPtr.h>
+
+namespace DGtal {
+  namespace functions {
+  /**
+   * Load existing look up table existing in file_name, precalculated
+   * tables can be accessed including the header:
+   * "DGtal/topology/tables/ConfigTables.h"
+   *
+   * @param input_filename plain text containing the bool table.
+   * @param known_size of the bitset, for 2D = 256 (2^8), 3D = 67108864 (2^26)
+   *
+   * @return smart ptr of map[neighbor_configuration] -> bool
+   *
+   * @note The tables were calculated using the generateTableXXX examples.
+   * Compressed files of the tables are distributed in the source code.
+   * At build or install time, the header
+   * "DGtal/topology/tables/ConfigTables.h" is generated.
+   * It has const strings variables with the file names of the tables.
+   */
+  inline
+  DGtal::CountedPtr< boost::dynamic_bitset<> >
```

Comment:
Cool, there is a dynamic_bitset now in C++. I had to write one in 2002, and the "not operation" was kind of a pain because of trailing bits.